convex/openai.ts

```diff
@@ -0,0 +1,69 @@
+"use node";
+
+import { action } from "./_generated/server";
+import { v } from "convex/values";
+import { api } from "../convex/_generated/api";
+import OpenAI from "openai";
+
+// Initialize the OpenAI client with the given API key
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY, // This is also the default, can be omitted
+});
+// start a game
+export const startGames = action({
+  args: {
+    topic: v.string(),
+    numberOfPlayer: v.number(),
+    roomId: v.id("rooms"),
+  },
+  handler: async (ctx, args) => {
+    try {
+      // ! TODO: check if this acutally works!!!
+      const checkTopicResponse = await openai.completions.create({
+        model: "text-davinci-003",
+        prompt: `if the  topic  '${args.topic}' is a good topic`,
+      });
+
+      if (checkTopicResponse.choices[0]?.text.includes("false")) {
+        return {
+          message: "The backend got an error",
+          error: "It's a bad topic!",
+        };
+      }
+
+      const response = await openai.completions.create({
+        model: "text-davinci-003",
+        prompt: `Suggest ${args.numberOfPlayer} words  for an topic that is '${args.topic}'`,
```

Comment:
I think this prompt is not so specific as it has to be. I just checked it in ChatGPT and the answer is not an array (as expected). You must be very specific, eg:
```javascript
const prompt = `Give me a list of ${args.nnumberOfPlayer} word related
to the topic "${args.topic}" as a json array format. Don't
include anything else in your response. The json array
must be as this: '{"words": ["word1", "word2", "word3"]}'.`
```

This prompt is just an example and we're not sure that it works. Maybe you need to test it in ChatGPT and show in this pr some examples of what ChatGPT returned. In your code, you will need to check if the response is in the expected format and try again if it fails. Limit the tries to 10 or something like that to avoid executing infinite calls to openai.

Reply:
This api I can't test, because I can't use it In China,So if you have the pay key, you can fix this problem

Reply:
You can test using ChatGPT from https://chat.openai.com/ , but I'll try to get a key